src/lsp/commands/vscode-commands.ts

```diff
@@ -0,0 +1,353 @@
+  const version = await downloader.ensureLSPServer(context, true);
+  void vscode.window.showInformationMessage(`Downloaded clojure-lsp version: ${version}`);
+};
+
+function configureTraceLogLevelHandler() {
+  void vscode.commands.executeCommand('workbench.action.openSettings', 'clojure.trace.server');
+}
+
+const manageHandler = async (
+  clients: defs.LSPClientMap,
+  start: StartHandler,
+  context: vscode.ExtensionContext
+) => {
+  const inactive_roots = await project_utils.findProjectRootsWithReasons().then((roots) => {
+    return filterOutRootsWithClients(roots, clients).map((root) => {
+      return {
+        label: `$(circle-outline) ${root.uri.path}`,
+        value: root.uri.path,
+        detail: root.reason,
+        active: false,
+      };
+    });
+  });
+
+  const active_roots = Array.from(clients.entries()).map(([key, client]) => {
+    let icon = '$(check)';
+    if (client.state === vscode_lsp.State.Starting) {
+      icon = '$(sync~spin)';
+    }
+
+    return {
+      label: `${icon} ${key}`,
+      value: key,
+      active: true,
+    };
+  });
+
+  type Choice = vscode.QuickPickItem & { value?: string; active?: boolean };
+  const choices: Choice[] = [];
+
+  if (active_roots.length > 0) {
+    choices.push(
+      {
+        label: 'Active',
+        kind: vscode.QuickPickItemKind.Separator,
+      },
+      ...active_roots
+    );
+  }
+
+  if (inactive_roots.length > 0) {
+    choices.push(
+      {
+        label: 'Inactive',
+        kind: vscode.QuickPickItemKind.Separator,
+      },
+      ...inactive_roots
+    );
+  }
+
+  choices.push(
+    {
+      label: 'General',
+      kind: vscode.QuickPickItemKind.Separator,
+    },
+    {
+      label: 'Open trace level settings',
+      value: '::trace-settings',
+    }
+  );
+  if (active_roots.length === 0) {
+    choices.push({
+      label: 'Download latest clojure-lsp version',
+      value: '::download',
+    });
+  }
+
+  const choice = await vscode.window.showQuickPick(choices, {
+    title: 'clojure-lsp',
+  });
+
+  if (!choice) {
+    return;
+  }
+
+  switch (choice.value) {
+    case '::download': {
+      void downloadServerHandler(context).catch((err) => console.error(err));
+      return;
+    }
+    case '::trace-settings': {
+      configureTraceLogLevelHandler();
+      return;
+    }
+  }
+
+  const action_choices: Array<vscode.QuickPickItem & { value: string }> = [];
+  if (choice.active) {
+    action_choices.push(
+      {
+        label: 'Stop',
+        value: '::stop',
+      },
+      {
+        label: 'Restart',
+        value: '::restart',
+      },
+      {
+        label: 'Show server info',
+        value: '::info',
+      },
+      {
+        label: 'Show server logs',
+        value: '::logs',
+      }
+    );
+  } else {
+    action_choices.push({
+      label: 'Start',
+      value: '::start',
+    });
+  }
+
+  const action = await vscode.window.showQuickPick(action_choices, { title: 'action' });
```

Comment:
Adds more context to the action menu title:

<img width="602" alt="image" src="https://user-images.githubusercontent.com/30010/213868585-946841a0-cc13-45ae-8e3a-e23de5a403a9.png">

```suggestion
  const action = await vscode.window.showQuickPick(action_choices, {
    title: `clojure-lsp: ${choice.value}`,
  });
```